### DISCUSSION TITLE
Granary getting goods - text

### DISCUSSION THREAD

**Discussion by catalin-duta** [Keriew/augustus#527] [Ideas]
I know that it's purely cosmetic, but would it make sense to have the granary say it's getting goods from another granary? Because really that's what it's doing.
![Screenshot_20211112-152855739](https://user-images.githubusercontent.com/93785920/141474980-83fb5121-63eb-41a4-ac01-4e9603a49f4d.jpg)

**Comment by Ouaz** [comment]
> but would it make sense to have the granary say it's getting goods from another granary?

Definitely. In fact, this same line (only in Augustus) is used both for warehouses and granaries, hence the error.

Another thing is that Granary special order is "getting goods" in Augustus, whereas it should be "getting food" like in Julius/C3 vanilla.

**Comment by Keriew** [comment]
Fixed both issues in dbb206f.

**Reply by Ouaz** [reply]
@Keriew 

Hello, this is Mr. Nitpicky here :yum: , but I never noticed that the food icon being collected by the granary's cartpusher is missing:

![granary_1](https://user-images.githubusercontent.com/11946570/144908257-36fcd5bd-f131-418a-843c-7be68648430a.png)
_Cartpusher on his way to get fish on another granary_

It actually appears only when the granary's cartpusher is on his way back:

![granary_2](https://user-images.githubusercontent.com/11946570/144908452-32e07318-e1bc-46b4-a79e-4855373beead.png)
_Cartpusher coming back with fish from another granary_

I compared with the same messages for warehouses, and the icon is displayed correctly for both:

![warehouse_get_goods_icon](https://user-images.githubusercontent.com/11946570/144908723-23b1daaf-8640-42c6-96d8-dbfb4847701a.png)
_Warehouseman on his way to get furniture from another warehouse_ (furniture icon displayed)

![warehouse_get_goods_icon_2](https://user-images.githubusercontent.com/11946570/144908899-9c76ba70-4703-4189-b891-85ac22c97c30.png)
Warehouseman coming back with furniture from another warehouse.

Sorry to ask you more work each time for tiny little things :blush:

**Reply by catalin-duta** [reply]
To pick another nit on top of Ouaz's:
Why are we getting *the* food or *the* goods?
It gives off a bit of a Mario vibe (I go-a gett-a da food-a)...

Also, maybe both carts are returning *with* a delivery is more accurate? Returning *from* a delivery sounds like "I've finished the delivery of goods and I'm coming back with an empty cart", like when a cart *delivers* to a workshop.

**Reply by Ouaz** [reply]
> Also, maybe both carts are returning _with_ a delivery is more accurate? Returning from a delivery sounds like "I've finished the delivery of goods and I'm coming back with an empty cart", like when a cart _delivers_ to a workshop.

Because it's the same line used for both. :yum: 

This line ("_Our cart is returning from a delivery_") is bugged in C3 vanilla (displayed all the time), whatever is currently doing the warehouseman, as the "delivery" line (although existing in the language file) is unused: ""_Our cart is taking goods elsewhere._"

Augustus fixed that recently (this past summer if I remember well), and added a line when warehousemen get foods/goods from another granary/warehouse (2 different lines since today).
But Augustus uses the same line than C3 vanilla when warehousemen come back, either they delivered something to somewhere or they get something from somewhere.

**Reply by Keriew** [reply]
> Hello, this is Mr. Nitpicky here 😋 , but I never noticed that the food icon being collected by the granary's cartpusher is missing:

That's because of the difference in how getting cartpushers work. The warehouse cartpusher chooses the resource to get when he is created, the granary cartpusher only decides when he reaches the destination granary. I guess we could change that.

I've implemented the rest of the suggestions in 8e91c22.

Don't feel bad about making suggestions like this, thank you for helping make the game better!
